### PR TITLE
Expand table-like input options for sphdistance

### PR DESCRIPTION
**Pull request by willschlitzer**
This pull request adds `data` to the parameter list in the docstring and moves the `sphdistance` function to the tabular data section in the index.

Fixes #

**Reminders**

- [ ] Run `make format` and `make check` to make sure the code follows the style guide.
- [ ] Add tests for new features or tests that would have caught the bug that you're fixing.
- [ ] Add new public functions/methods/classes to `doc/api/index.rst`.
- [ ] Write detailed docstrings for all functions/methods.
- [ ] If adding new functionality, add an example to docstrings or tutorials.

**Slash Commands**

You can write slash commands (`/command`) in the first line of a comment to perform
specific operations. Supported slash commands are:

- `/format`: automatically format and lint the code
- `/test-gmt-dev`: run full tests on the latest GMT development version

### CODE REVIEW

File: pygmt/src/sphdistance.py [hunk 1]
```diff
@@ -35,6 +35,10 @@ def sphdistance(table, **kwargs):
 
     Parameters
     ----------
+    table : str or {table-like}
```
Comment (weiji14) [inline pygmt/src/sphdistance.py:38]: Wait for more votes to come in on whether to use `data` or `table` at https://github.com/GenericMappingTools/pygmt/issues/1479#issuecomment-912203679 :slightly_smiling_face:

```suggestion
    datatable : str or {table-like}
```

Comment (michaelgrund) [reply pygmt/src/sphdistance.py:38]: Voted in #1479 😉

Comment (weiji14) [reply pygmt/src/sphdistance.py:38]: Ok, looks like we're going with `data`! Please apply the change.

**Review by weiji14 — COMMENTED** [review]
Could you also:

- Rename `table` to `data` at L25, L61, and also in `pygmt/tests/test_sphdistance.py`
- At L52: Insert blank line between:

```
       Return type depends on whether the ``outgrid`` parameter is set:
        - :class:`xarray.DataArray` if ``outgrid`` is not set
```
to prevent a Sphinx warning `pygmt/pygmt/src/sphdistance.py:docstring of pygmt.src.sphdistance.sphdistance:67: WARNING: Unexpected indentation`

**Comment by seisman** [comment]
Do you want to add "xyz" input support (like #1531) in this PR or a separate PR?

**Comment by willschlitzer** [comment]
> Do you want to add "xyz" input support (like #1531) in this PR or a separate PR?

I can give it a try.

File: pygmt/src/sphdistance.py [hunk 2]
```diff
@@ -22,7 +22,7 @@
     V="verbose",
 )
 @kwargs_to_strings(I="sequence", R="sequence")
-def sphdistance(table, **kwargs):
+def sphdistance(x=None, y=None, z=None, data=None, **kwargs):
```
Comment (seisman) [inline pygmt/src/sphdistance.py:25]: As discussed in #1479, the order should be `data, x, y, z`:
```suggestion
def sphdistance(data=None, x=None, y=None, z=None, **kwargs):
```

File: pygmt/src/sphdistance.py [hunk 3]
```diff
@@ -35,6 +35,12 @@ def sphdistance(table, **kwargs):
 
     Parameters
     ----------
+    x/y/z : 1d arrays
+        Arrays of x and y coordinates and values z of the data points.
+    data : str or {table-like}
+        Pass in (x, y, z) or (longitude, latitude, elevation) values by
+        providing a file name to an ASCII data table, a 2D
+        {table-classes}.
```
Comment (seisman) [inline pygmt/src/sphdistance.py:43]:
```suggestion
    data : str or {table-like}
        Pass in (x, y, z) or (longitude, latitude, elevation) values by
        providing a file name to an ASCII data table, a 2D
        {table-classes}.
    x/y/z : 1d arrays
        Arrays of x and y coordinates and values z of the data points.
```

File: pygmt/src/sphdistance.py [hunk 4]
```diff
@@ -54,7 +61,9 @@ def sphdistance(table, **kwargs):
         raise GMTInvalidInput("Both 'region' and 'spacing' must be specified.")
     with GMTTempFile(suffix=".nc") as tmpfile:
         with Session() as lib:
-            file_context = lib.virtualfile_from_data(check_kind="vector", data=table)
+            file_context = lib.virtualfile_from_data(
+                check_kind="vector", data=data, x=x, y=y, z=z
```
Comment (seisman) [inline pygmt/src/sphdistance.py:65]:
```suggestion
                check_kind="vector", data=data, x=x, y=y, z=z, required_z=True
```

**Comment by willschlitzer** [comment]
@seisman Doesn't `sphdistance` only require lat-long inputs? I don't think there should be `z` inputs.

**Comment by seisman** [comment]
> @seisman Doesn't `sphdistance` only require lat-long inputs? I don't think there should be `z` inputs.

I've never used `sphdistance`, but reading the sphdistance manpage, I think only `x` and `y` are required.

**Comment by willschlitzer** [comment]
> > @seisman Doesn't `sphdistance` only require lat-long inputs? I don't think there should be `z` inputs.
> 
> I've never used `sphdistance`, but reading the sphdistance manpage, I think only `x` and `y` are required.

So should I then remove `z` from the parameters? I'm getting this error when I run the tests: `pygmt.exceptions.GMTInvalidInput: data must provide x, y, and z columns`

**Comment by seisman** [comment]
> > > @seisman Doesn't `sphdistance` only require lat-long inputs? I don't think there should be `z` inputs.
> > 
> > 
> > I've never used `sphdistance`, but reading the sphdistance manpage, I think only `x` and `y` are required.
> 
> So should I then remove `z` from the parameters? I'm getting this error when I run the tests: `pygmt.exceptions.GMTInvalidInput: data must provide x, y, and z columns`

Yes, please. Sorry for misleading you.

File: pygmt/src/sphdistance.py [hunk 5]
```diff
@@ -35,6 +35,12 @@ def sphdistance(table, **kwargs):
 
     Parameters
     ----------
+    data : str or {table-like}
+        Pass in (x, y) or (longitude, latitude) values by
+        providing a file name to an ASCII data table, a 2D
+        {table-classes}.
+    x/y/z : 1d arrays
+        Arrays of x and y coordinates and values z of the data points.
```
Comment (seisman) [inline pygmt/src/sphdistance.py:43]:
```suggestion
    x/y : 1d arrays
        Arrays of x and y coordinates.
```

**Review by seisman — APPROVED** [review]

File: pygmt/src/sphdistance.py [hunk 6]
```diff
@@ -22,7 +22,7 @@
     V="verbose",
 )
 @kwargs_to_strings(I="sequence", R="sequence")
-def sphdistance(table, **kwargs):
+def sphdistance(data=None, x=None, y=None, **kwargs):
     r"""
     Create Voroni polygons from lat/lon coordinates.
```
Comment (weiji14) [inline pygmt/src/sphdistance.py:27]: Typo here. Also, just noticed that this summary doesn't seem correct. It seems to suggest Voronoi polygons are created, but the output is a raster grid, not a vector polygon.
```suggestion
    Create Voronoi polygons from lat/lon coordinates.
```

File: pygmt/src/sphdistance.py [hunk 7]
```diff
@@ -35,6 +35,12 @@ def sphdistance(table, **kwargs):
 
     Parameters
```
Comment (weiji14) [inline pygmt/src/sphdistance.py:36]: Can't make a suggested change here, but could you include a line:

```
Full option list at :gmt-docs:`sphdistance.html
```
above the `{aliases}` line? Thanks!

**Review by weiji14 — COMMENTED** [review]
The docstring description reads: `sphdistance reads one or more ASCII [or binary] files (or standard input)`, but 'standard input' (stdin) doesn't make sense in the Python world, suggest rewording it to use just 'table', similar to https://github.com/GenericMappingTools/pygmt/pull/1418#discussion_r716167580.

**Comment by seisman** [comment]
@willschlitzer The tests fail because the Windows server cannot download the cache file `EGM96_to_36.txt`, which was added in #1434, but not cached.

Please trigger the `cache_data.yml` workflow manually to update the caches.

**Comment by willschlitzer** [comment]
> Please trigger the `cache_data.yml` workflow manually to update the caches.

@seisman How do I do this?

**Comment by seisman** [comment]
> > Please trigger the `cache_data.yml` workflow manually to update the caches.
> 
> @seisman How do I do this?

You need to uncomment the following line (removing the leading `#`), commit the changes, and wait for the finish of the workflow. After that, you need to revert that commit:

https://github.com/GenericMappingTools/pygmt/blob/0926206bf81e7af86dd3d84da82c38f70f13fe5d/.github/workflows/cache_data.yaml#L6

**Comment by willschlitzer** [comment]
> You need to uncomment the following line (removing the leading `#`), commit the changes, and wait for the finish of the workflow. After that, you need to revert that commit:

Likely won't get to this until tomorrow morning (UK time)

File: .github/workflows/cache_data.yaml [hunk 8]
```diff
@@ -3,7 +3,7 @@ name: Cache data
 
 on:
   # Uncomment the 'pull_request' line below to manually re-cache data artifacts
-  # pull_request:
+  pull_request:
```
Comment (weiji14) [inline .github/workflows/cache_data.yaml:6]:
```suggestion
  # pull_request:
```

File: pygmt/src/sphdistance.py [hunk 9]
```diff
@@ -22,19 +22,27 @@
     V="verbose",
 )
 @kwargs_to_strings(I="sequence", R="sequence")
-def sphdistance(table, **kwargs):
+def sphdistance(data=None, x=None, y=None, **kwargs):
     r"""
-    Create Voroni polygons from lat/lon coordinates.
+    Create Voronoi polygons from lat/lon coordinates.
```
Comment (weiji14) [inline pygmt/src/sphdistance.py:27]:
```suggestion
    Create Voronoi distance, node, or natural nearest-neighbor grid on a
    sphere.
```

**Review by weiji14 — APPROVED** [review]
